apis/Google.Cloud.Firestore.Data/Google.Cloud.Firestore.Data.IntegrationTests/FirestoreFixture.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright 2017, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using Xunit;
+
+namespace Google.Cloud.Firestore.Data.IntegrationTests
+{
+    [CollectionDefinition(nameof(FirestoreFixture))]
+    public class FirestoreFixture : IDisposable, ICollectionFixture<FirestoreFixture>
+    {
+        // This is not the test project environment variable used by other integration tests,
+        // as Datastore and Firestore can't both be active in the same project.
+        private const string ProjectEnvironmentVariable = "FIRESTORE_TEST_PROJECT";
+        public FirestoreDb FirestoreDb { get; }
+
+        /// <summary>
+        /// A randomly generated prefix for collections.
+        /// </summary>
+        public string CollectionPrefix { get; }
+
+        /// <summary>
+        /// A collection intended for tests to create and fetch documents in. Don't query in tests!
+        /// </summary>
+        public CollectionReference NonQueryCollection { get; }
+
+        public FirestoreFixture()
+        {
+            string projectId = Environment.GetEnvironmentVariable(ProjectEnvironmentVariable);
+            if (string.IsNullOrEmpty(projectId))
+            {
+                throw new InvalidOperationException(
+                    $"Please set the {ProjectEnvironmentVariable} environment variable before running tests");
+            }
+
+            // Currently, only the default database is supported... so we create all our collections with a randomly-generated prefix.
+            // When multiple databases are supported, we'll create a new one per test run.
+            CollectionPrefix = Guid.NewGuid().ToString();
+            FirestoreDb = FirestoreDb.Create(projectId);
+            NonQueryCollection = FirestoreDb.Collection(CollectionPrefix + "-non-query");
+        }        
+
+        public void Dispose()
+        {
+            // No-op for now. With multiple-database support, we'll 
```

Comment:
Such a cliff-hanger...
We'll *what*?!?

Reply:
Whoops again. I've clearly got too many things going on...